### PR TITLE
[WIP] superenv:  Don't blacklist XQuartz for formulas which require it.

### PR DESCRIPTION
**Pull request by RandomDSdevel**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'What it says on the tin,' as the saying goes.  

Closes Homebrew/homebrew-core#30949.  

-----

Outstanding Tasks:  

* [ ] Replace with a more correct approach, if necessary, as this likely isn't entirely the best way to do things here.  
* [ ] Address review comments (both mine and others' when they arrive.)  
* [ ] Write tests.

### CODE REVIEW

File: Library/Homebrew/shims/super/cc [hunk 1]
```diff
@@ -238,6 +238,11 @@ class Cmd
     elsif path.start_with?(prefix, tmpdir)
       true
     elsif path.start_with?("/opt/local", "/opt/boxen/homebrew", "/opt/X11", "/sw", "/usr/X11")
+      # Don't ignore either `/opt/x11` or `/usr/x11` for formulas which `depends_on :x11`:
+      if path.start_with?("/opt/X11", "/usr/X11")
+         && # `formula.requirements.include?(:x11)` or similar, but I don't know how to hook this up
```
Comment (RandomDSdevel) [inline Library/Homebrew/shims/super/cc]: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;How would I pass this through into this file and access it here?

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The incomplete part of this PR is probably what's causing the test failures; I'll fix it once I know how to flesh that out.

File: Library/Homebrew/shims/super/cc (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/shims/super/cc]: It's not possible to access this information at superenv time.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Closed in favor of other work on Homebrew/homebrew-core#30949.

**Pull request closed without merging**